### PR TITLE
[INFINITY-1811] Enable resource refinement

### PR DESCRIPTION
**Pull request by gabrielhartmann**
This is the last PR related to the implementation of resource refinement.  It depends on `testing/pull/1524` being merged in order for the integration test to pass.

CC: @mpark

### CODE REVIEW

File: frameworks/helloworld/tests/test_resource_refinement.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def teardown_module(module):
+    install.uninstall(PACKAGE_NAME)
+
+
+@pytest.mark.sanity
+@pytest.mark.smoke
+def test_install():
+    check_running(PACKAGE_NAME)
```
Comment (nickbp) [inline frameworks/helloworld/tests/test_resource_refinement.py]: Any kind of straightforward test that could be run to validate that something is visible on the public node? (feel free to defer, or skip if not relevant)

Comment (gabrielhartmann) [inline frameworks/helloworld/tests/test_resource_refinement.py]: There isn't anything being served there.  It's just one instance of a hello pod.  This isn't really specific to `slave_public` anyway.  It's just a test to see if we can consume statically reserved resources.  `slave_public` happens to be the most convenient and the one needed by EdgeLB.

Comment (nickbp) [inline frameworks/helloworld/tests/test_resource_refinement.py]: :+1:

Comment (gabrielhartmann) [inline frameworks/helloworld/tests/test_resource_refinement.py]: A more thorough test would be one that gets the TaskInfo from the scheduler and validates that the single hello pod has the expected roles set.  Or you could probably get that from a Mesos endpoint.

**Comment by gabrielhartmann** [comment]
retest this please

**Comment by gabrielhartmann** [comment]
retest this please

**Comment by benclarkwood** [comment]
retest this please (RR landed in dc/os)

**Comment by nickbp** [comment]
Created a `testing/master` cluster and was able to manually deploy against `examples/pre-reserved.yml` (i.e. the test case that was added) just fine. I was also able to do a default `svc.yml` hello-world install just fine on the same cluster.

**Comment by nickbp** [comment]
retest this please (ccm)

**Comment by nickbp** [comment]
OK I'm pretty sure something's going on. For example, in other PRs kafka takes 30 minutes to finish, while it's consistently 2h+ (and counting) here...

**Comment by benclarkwood** [comment]
:(

**Comment by benclarkwood** [comment]
It appears the kafka tests have passed, however?

**Comment by nickbp** [comment]
Oh huh. Well that's really odd then. I was assuming they were going to fail.

**Comment by gabrielhartmann** [comment]
@nickbp and @benclarkwood: Maybe figure out what clusters this run is on, bump their lifespans in CCM substantially, let the tests complete, but don't merge so you can get the test output for Kafka etc. and see what's taking so long.

**Comment by nickbp** [comment]
Added 8 hours to `infinity-test-6odpyzbk` + `infinity-test-jo5q4k55`

**Comment by nickbp** [comment]
Looks like kafka didn't pass, wonder why CI thinks it succeeded? @jsrodman 

from https://jenkins.mesosphere.com/service/jenkins/job/dcos-commons/job/inf-pr/4491/consoleFull:
```
 generated xml file: /jenkins/workspace/dcos-commons/inf-pr/kafka-shakedown-report.xml 
=================================== FAILURES ===================================
_________________________ test_pods_restart_on_overlay _________________________
...
_______________________ test_port_static_to_static_port ________________________
...
_______________________ test_port_static_to_dynamic_port _______________________
...
many more...
...
============================== 1 tests deselected ==============================
============ 14 failed, 21 passed, 1 deselected in 9234.71 seconds =============
Marathon apps: []
[GH-STATUS]  success: Shakedown tests succeeded
Deleting CLI sandbox: /tmp/ci-test-2errk20m
```

**Comment by nickbp** [comment]
HDFS also failed, with a possible common thread of tests which exercise the overlay:

```
==================================== ERRORS ====================================
___________________ ERROR at setup of test_tasks_on_overlay ____________________
....
============================== 2 tests deselected ==============================
============= 3 skipped, 2 deselected, 16 error in 1873.60 seconds =============
[GH-STATUS]  success: Shakedown tests succeeded
```

**Comment by gabrielhartmann** [comment]
@benclarkwood:  Thanks for the change.  LGTM

**Pull request merged**